### ISSUE TITLE
Use the d3js library to jazz up the exams page?

### ISSUE THREAD

**Issue by WillThompson** [wayneyu/merapp#118]

**Comment by WillThompson** [comment]
i.e.
Compare this:
http://merapp.herokuapp.com/exams
and this:
http://merapp.herokuapp.com/topics

The exams page looks pretty bland by comparison.

**Comment by BernhardKonrad** [comment]
Hm, do you have an idea of how we could improve the exams page? Maybe the d3 gallery could be a source of inspiration: https://github.com/mbostock/d3/wiki/Gallery
Eg, how about something like this [BubbleMenu](http://sunsp.net/demo/BubbleMenu/)?

**Comment by WillThompson** [comment]
I like the Bubblemenu, but would it make sense to use the same presentation as used for the topics for consistency? (with the area of each circle approximately proportional to the number of completed exams)

**Comment by WillThompson** [comment]
Actually, I think that since we have a number of parent topics... which make it hard to see which element to click on to reach a particular topic... it may make sense to use  

http://mbostock.github.io/d3/talk/20111018/tree.html

for the topics menu as well as the exams menu:

What do you think?

**Comment by BernhardKonrad** [comment]
We could do that too. But the topic zoom map has no natural ordering (just as the topics), whereas the courses do. So I don't think we need to be consistent in this case.

**Comment by BernhardKonrad** [comment]
The tree for the topics would also work. Would be nice to have them side-by-side. Or maybe use this for the topics: http://bl.ocks.org/mbostock/1093025

**Comment by WillThompson** [comment]
I think either of them would be good. I personally like the tree, but I don't feel attached to a particular one.

**Comment by BernhardKonrad** [comment]
The current topic zoom map could also be improved: eg on hover() of a parent topic we could display all subtopic in a side panel.

**Comment by WillThompson** [comment]
I'm not concerned about that... I'm concerned that if a student clicks on a topic that has a large number of children that they may not be sure which topic they're clicking on.
e.g. If we click on "Elementary number theory" on the topics page ( http://merapp.herokuapp.com/topics ), then we suddenly see ~10 topics with long names clustered together in the centre and it becomes difficult to click on the one of interest.

**Comment by BernhardKonrad** [comment]
Yes, that's certainly a problem. I think this can be rescued by highlighting the hover subtopic. But if you think it's better to replace the zoom map altogether, go for it! I'm happy to help with an intro to d3.
